src/foam/nanos/ruler/Rule.js

```diff
@@ -380,8 +386,8 @@
       name: 'javaExtras',
       buildJavaClass: function(cls) {
         cls.extras.push(`
-        public static Rule findById(Collection<Rule> listRule, String passedId) {
-          return listRule.stream().filter(rule -> passedId.equals(rule.getId())).findFirst().orElse(null);
+        public static Rule findById(Collection<Rule> listRule, Long id) {
```

Comment:
Why Long instead of long?

Reply:
Where is this called from? Is it performance sensitive? If so, could you use a Map instead?

Reply:
i think this is called from a listener on ruledao in the rulerdao decorator to add to or remove from the rules list of the target dao whenever the ruledao changes